keras_retinanet/preprocessing/df_generator.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+
+from keras_retinanet.preprocessing.csv_generator import CSVGenerator, Generator
+
+
+class DfGenerator(CSVGenerator):
+    """Custom generator intented to work with in-memory Pandas' dataframe."""
+    def __init__(self, df, class_mapping, cols, base_dir='', **kwargs):
+        """Initialization method.
+
+        Arguments:
+            df: Pandas DataFrame containing paths, labels, and bounding boxes.
+            class_mapping: Dict mapping label_str to id_int.
+            cols: Dict Mapping 'col_{filename/label/x1/y1/x2/y2} to corresponding df col.
+        """
+        self.base_dir = base_dir
+        self.cols = cols
+        self.classes = class_mapping
+        self.labels = {v: k for k, v in self.classes.items()}
+
+        self.image_data = self._read_data(df)
+        self.image_names = list(self.image_data.keys())
+
+        Generator.__init__(self, **kwargs)
```

Comment:
Prefer to use `super(CSVGenerator, self).__init__(self, **kwargs)` instead. Assuming your intention is to skip the CSVGenerator initializer and initialize the `Generator` instead.

Reply:
Why does this inherit from `CSVGenerator` at all? That seems unnecessary, and may prevent us from changing internal details of the `CSVGenerator` in the future.